gcc/testsuite/gcc.target/riscv/cv-bi-beqimm-compile-1.c

```diff
@@ -0,0 +1,16 @@
+/* { dg-do compile } */
+/* { dg-options "-march=rv32i_xcvbi1p0 -mabi=ilp32" } */
+/* { dg-skip-if "" { *-*-* }  { "-O0" } { "" } } */
+
+/* __builtin_expect is used to provide the compiler with branch prediction information.
+   For example, __builtin_expect(a, 10), will expect "a" to be 10.
+   This gives a hint to the compiler what the value of "a" might be,
+   using this information it will optimised the best path. */
```

Comment:
If you are going to put his comment here, you should put in *all* the tests where it is used.  You can't rely on someone looking at a different test to look out the comments in other ones.

However I think you can reply on the reader knowing what `__builtin_expect` does, so you could simplify the comment to something like:

> _builtin_expect is used to direct the compiler to the expected flow though the code, and hence selection of the appropriate instruction.

Reply:
I have updated the comment, and put them in all relevant tests. Thank you for the feedback.